test_plans/queue_constructors.asciidoc

```diff
@@ -0,0 +1,82 @@
+* `queue(const async_handler& asyncHandler,
+               const property_list& propList = {})`
+* `queue(const DeviceSelector& deviceSelector,
+               const property_list& propList = {})`
+* `queue(const DeviceSelector& deviceSelector,
+               const async_handler& asyncHandler,
+               const property_list& propList = {})`
+* `queue(const device& syclDevice, const property_list& propList = {})`
+* `queue(const device& syclDevice, const async_handler& asyncHandler,
+               const property_list& propList = {})`
+* `queue(const context& syclContext, const DeviceSelector& deviceSelector, const property_list& propList = {})`
+* `queue(const context& syclContext, const DeviceSelector& deviceSelector,
+               const async_handler& asyncHandler, const property_list& propList = {})`
+* `queue(const context& syclContext, const device& syclDevice, const property_list& propList = {})`
+* `queue(const context& syclContext, const device& syclDevice,
+               const async_handler& asyncHandler,
+               const property_list& propList = {})`
+
+If `device::has(sycl::aspect::usm_host_allocations)` is `false` this part of test is skipped.
+Submit two `single_task` to queue - first with long loop and second that changes a `usm` variable with host allocation.
+In the first task check that after loop the variable is unchanged to verify that `in_order` property adds the requirement that the SYCL queue provides in-order semantics.
+
+=== Check constructors with both properties
+
+Use `propList = {in_order, enable_profiling}` with following constructors also using selected device, corresponding context, `cts_selector`, `cts_async_handler` and check that resulting queue member finctions `has_property<enable_profiling>()`, `has_property<in_order>()` and `is_in_order()` return `true`:
+
+* `queue(const property_list& propList = {})`
+* `queue(const async_handler& asyncHandler,
+               const property_list& propList = {})`
+* `queue(const DeviceSelector& deviceSelector,
+               const property_list& propList = {})`
+* `queue(const DeviceSelector& deviceSelector,
+               const async_handler& asyncHandler,
+               const property_list& propList = {})`
+* `queue(const device& syclDevice, const property_list& propList = {})`
+* `queue(const device& syclDevice, const async_handler& asyncHandler,
+               const property_list& propList = {})`
+* `queue(const context& syclContext, const DeviceSelector& deviceSelector, const property_list& propList = {})`
+* `queue(const context& syclContext, const DeviceSelector& deviceSelector,
+               const async_handler& asyncHandler, const property_list& propList = {})`
+* `queue(const context& syclContext, const device& syclDevice, const property_list& propList = {})`
+* `queue(const context& syclContext, const device& syclDevice,
+               const async_handler& asyncHandler,
+               const property_list& propList = {})`
```

Comment:
Do we already have a test of the profiling APIs?  If we do not already test this, can we add a simple test like this:

* Check if the selected device has `aspect::queue_profiling`.  Skip this test if it does not.
* Construct a `queue` from the selected device with the `enable_profiling` property.
* Submit a simple command with `single_task`, saving the returned `event`.
* Wait for the task to complete.
* Get the following timestamp values from the `event`:
   * `submit = event::get_profiling_info<info::event_profiling::command_submit>()`
   * `start = event::get_profiling_info<info::event_profiling::command_start>()`
   * `end = event::get_profiling_info<info::event_profiling::command_end>()`
* Check that:
   * `end >= start`
   * `start >= submit`

Reply:
It is tested here https://github.com/KhronosGroup/SYCL-CTS/blob/SYCL-2020/tests/event/event.cpp#L450